hosted-cluster-config-operator/operator/config.go

```diff
@@ -93,9 +97,6 @@ func (c *HostedClusterConfigOperatorConfig) Manager() ctrl.Manager {
 func mgr(cfg *rest.Config) ctrl.Manager {
 	cfg.UserAgent = "hosted-cluster-config-operator-manager"
 	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
-		LeaderElection:          true,
```

Comment:
Why are we removing the leader election? Wont that result in two hccos racing with each other during an upgrade?

Reply:
We should use a recreate strategy for the deployment. This is not something that is required to be running all the time. Will fix.

Reply:
But even with the recreate startegy kube doesn't guarantee that there aren't two pods running at the same time, its a best-efford thing..What is the drawback of the leaderelection that made you remove it?

Reply:
We were using the guest cluster client for the leader election, and in some cases it would take a long time to become a leader. One different possible fix is to use a lease resource on the control plane side. Will take a look.